tl.tl

```diff
@@ -4663,6 +4715,26 @@ parse_statements = function(ps: ParseState, i: integer, toplevel?: boolean): int
       i, item = fn(ps, i)
 
       if item then
+         -- store unattached comments
+         if toplevel and token.comments then
+            for _, tc in ipairs(token.comments) do
+               local is_attached = false
+               if item.comments then
+                  for _, nc in ipairs(item.comments) do
+                     if tc == nc then
+                        is_attached = true
+                        break
+                     end
+                  end
+               end
+               if not is_attached then
+                  if not unattached_comments then
+                     unattached_comments = {}
+                  end
+                  table.insert(unattached_comments, tc)
+               end
```

Comment:
`else break` here too? Once you find one attached comment, all subsequent ones are attached too, right?